Sources/Views/MessageInputBar.swift

```diff
@@ -443,6 +443,8 @@ open class MessageInputBar: UIView {
     /// - Parameter positions: The UIStackView's to layout
     public func layoutStackViews(_ positions: [InputStackView.Position] = [.left, .right, .bottom, .top]) {
         
+        guard superview != nil else { return }
+        
```

Comment:
We already check the superview at the top of the function so we don't need to check again

Reply:
@nathantannar4 , sorry , my mistake the comments I make before. I would update the code.

Reply:
@nathantannar4 , Hmm AutoLayout issues still existed when remove redundant `superview` check as you say. 

we can reproduce it when MessageInputBar become first responder, and then switch the keyboard.

```
Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
2017-11-06 14:55:39.752989+0800 ChatExample[3284:1031668] [LayoutConstraints] Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. 
	Try this: 
		(1) look at each constraint and try to figure out which you don't expect; 
		(2) find the code that added the unwanted constraint or constraints and fix it. 
(
    "<NSLayoutConstraint:0x600000298d30 MessageKit.InputTextView:0x7fc9d1137200.bottom == MessageKit.InputStackView:0x7fc9d056d410.top   (active)>",
    "<NSLayoutConstraint:0x60000028b680 MessageKit.InputStackView:0x7fc9d056d410.bottom == UILayoutGuide:0x6000007aa8e0'UIViewSafeAreaLayoutGuide'.bottom - 6   (active)>",
    "<NSLayoutConstraint:0x600000289c40 V:|-(0)-[MessageKit.InputStackView:0x7fc9d058ab60]   (active, names: '|':MessageKit.MessageInputBar:0x7fc9d0565f70 )>",
    "<NSLayoutConstraint:0x600000288340 MessageKit.InputStackView:0x7fc9d058ab60.bottom == MessageKit.InputTextView:0x7fc9d1137200.top - 6   (active)>",
    "<NSLayoutConstraint:0x600000288480 '_UITemporaryLayoutHeight' MessageKit.MessageInputBar:0x7fc9d0565f70.height == 0   (active)>",
    "<NSLayoutConstraint:0x60000028af50 'UIViewSafeAreaLayoutGuide-bottom' V:[UILayoutGuide:0x6000007aa8e0'UIViewSafeAreaLayoutGuide']-(0)-|   (active, names: '|':MessageKit.MessageInputBar:0x7fc9d0565f70 )>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x60000028b680 MessageKit.InputStackView:0x7fc9d056d410.bottom == UILayoutGuide:0x6000007aa8e0'UIViewSafeAreaLayoutGuide'.bottom - 6   (active)>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
The methods in the UIConstraintBasedLayoutDebugging category on UIView listed in <UIKit/UIView.h> may also be helpful.
```

Reply:
None of these should happen when using the API correctly, even without the superview check. Are you animating? All of the examples in the Example project do not get these issues so I am curious as to what you are doing in the code? You should not be making the input bar the first responder manually. iOS takes care of that because it’s an input access view

Reply:
Reproduce steps:
![messageinputbarbug](https://user-images.githubusercontent.com/5061845/32472739-fca25704-c39e-11e7-8272-a3662131c51c.gif)

Reply:
@zhongwuzw I don't get the same errors under the latest master branch. Did you adjust anything in the ConversationViewController? I am sorry for being picky

Reply:
@nathantannar4 , nope, I downloaded project from `master` directly without any modification, it can reproduce in Simulator or Device.